internal/pkg/java/server.go

```diff
@@ -146,6 +155,20 @@ func (s Server) ProcessConn(c net.Conn, webhooks []webhook.Webhook) (infrared.Co
 		return infrared.ConnTunnel{}, err
 	}
 
+	if s.GetSendProxyProtocol() {
+		header := &proxyproto.Header{
+			Version:           2,
+			Command:           proxyproto.PROXY,
+			TransportProtocol: proxyproto.TCPv4,
+			SourceAddr:        pc.RemoteAddr(),
+			DestinationAddr:   rc.RemoteAddr(),
+		}
+
+		if _, err := header.WriteTo(&rc); err != nil {
```

Comment:
You'd probably want to use `rc.w` here. Referencing an infrared Conn object that's unknown to the proxyprotocol library could lead to segfaults.

Reply:
Yes and no. Conn implements the io.Writer interface via w https://github.com/haveachin/infrared/blob/bee5cb005780b6d2d63a0efc00d0b85f661907d5/internal/pkg/java/conn.go#L47-L49